src/compiler/checker.ts

```diff
@@ -13175,6 +13177,21 @@ namespace ts {
             if (node.flags & NodeFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            checkNameOfStaticMethodOrPropertyInClass(node);
+        }
+
+        function checkNameOfStaticMethodOrPropertyInClass(node: ClassElement) {

```

Comment:
@mhegazy and @yuit , thanks for the review, I will find time to fix these.

I'm wondering how to use the `Function` type, `name === nameof Function.length` is not allowed yet, and `Function[name] !== undefined` will not work for the case of `name === "apply"`.